google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -217,6 +222,26 @@ def bq_to_arrow_schema(bq_schema):
     return pyarrow.schema(arrow_fields)
 
 
+def bq_schema_to_nullsafe_pandas_dtypes(bq_schema: Sequence[schema.SchemaField]):
+    """Return the default dtypes to use for columns in a BigQuery schema.
+
+    Only returns default dtypes which are safe to have NULL values. This
+    includes Int64, which has pandas.NA values and does not result in
+    loss-of-precision.
+
+    Returns:
+        Dict[str, str]: mapping from column names to dtypes
+    """
+    dtypes = {}
+    for bq_field in bq_schema:
+        if bq_field.mode.upper() not in {"NULLABLE", "REQUIRED"}:
+            continue
```

Comment:
TODO(tswast): Need unit tests for this line.

Reply:
Done in https://github.com/googleapis/python-bigquery/pull/786/commits/8f90c511d9e97c0341352b71ee4499d886c9c78d